core-bundle/src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -410,7 +410,7 @@ public function showAll()
 				while ($objRoot->next())
 				{
 					// Respect given filemounts, e.g. set in the initPicker() method
-					if (!empty($this->arrFilemounts) && !$isBasePath($objRoot->path, $this->arrFilemounts))
+					if ((!empty($this->arrFilemounts) || null !== $root) && !$isBasePath($objRoot->path, $this->arrFilemounts))
```

Comment:
This is needed to make sure the search cannot escape the path if `$this->arrFilemounts` was set to an empty array because of no mount matched the path.